### DISCUSSION TITLE
How to use Prisma Client for the best performance?

### DISCUSSION THREAD

**Discussion by gustavo-maurina** [prisma/prisma#11891] [Q&A] [answered]
I have a couple questions on what is the best way to use Prisma in a project to get the best performance.

- Should I declare a `PrismaClient` instance somewhere and use it all over the project or declare a new instance on every file?
- Should I use `$disconnect()` method after using PrismaClient?

These are my question, I'd like to hear the different approaches and their pros and cons on these topics. 
Thanks in advance!

**Comment by nurul3101** [comment, marked as answer]
Hey @gustavo-maurina 👋🏻 

- Your application should generally only create **one** instance of `PrismaClient`. The reason for this is that each instance of PrismaClient manages a connection pool, which means that a large number of clients can exhaust the database connection limit.
You can get more information about number of PrismaClient Instances and how to achieve it [here](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/instantiate-prisma-client#the-number-of-prismaclient-instances-matters)
- Explicitly calling `$disconnect` method makes sense only if Your program runs infrequently (for example, a scheduled job to send emails each night), which means it does not benefit from a long-running connection to the database. If you `$disconnect` after every request in your REST or GraphQL server then every subsequent request would take time to establish a connection again, and doing so as part of each request will slow down your application.

Please let me know if you have any other questions.

**Reply by gustavo-maurina** [reply]
Very clarifying, thanks a lot!

**Reply by k4nedito** [reply]
I randomly had two very specific questions and you literally answered them in the most detailed way possible... 1 year ago

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
